app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -31,37 +31,39 @@ public Cursor getTalkAdapterData(
             boolean isDownloaded
     )
     {
-        List<String> columns = new ArrayList<>();
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.ID);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
-        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
-        return getTalks(columns, searchTerms, isStarred, isDownloaded);
+
+        return getTalks(searchTerms, isStarred, isDownloaded, "");
     }
 
     /**
-     * @param columns the columns to select or null if all
      * @param searchTerms the term to filter by or null if none
      * @param isStarred whether the talk is starred
      * @param isDownloaded whether the talk is downloaded
+     * @param extraWhere an extra where clause that can be used to filter on a specific teacher/center id
      * @return every talk in the db filtered by search term, is starred, and is downloaded
      */
     private Cursor getTalks(
-           List<String> columns,
            List<String> searchTerms,
            boolean isStarred,
-           boolean isDownloaded
+           boolean isDownloaded,
+           String extraWhere
     )
     {
+        List<String> columns = new ArrayList<>();
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
+        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
+        columns.add(DBManager.C.Center.TABLE_NAME    + "." + DBManager.C.Center.NAME);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.DURATION_IN_MINUTES);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.RECORDING_DATE);
+
         String queryColumns = "";
-        if (columns == null || columns.size() == 0)
-        {
-            queryColumns = "*";
-        }
-        else
-        {
-            queryColumns = TextUtils.join(", ", columns);
+        List<String> named_columns = new ArrayList<String>();
+        named_columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID);
+        for(String c : columns) {
+            named_columns.add(c + " AS " + c.replace(".", "_"));
```

Comment:
This was necessary because `SQLiteCursor` ignores table name qualifiers on the returned columns, so it's not possible to distinguish between things like `teacher.name` and `center.name`.  It also doesn't seem legal to have a `.` in the alias name, which is why I used `_` instead.  Not sure if there's a better way to do this...

Reply:
That sounds great!  Thanks for taking the time to make unit tests, it'll be awesome to have those in place. :)